spellbook/models/x2y2/ethereum/x2y2_ethereum_events.sql

```diff
@@ -0,0 +1,180 @@
+    FROM direct_x2y2_txs
+    )
+
+, all_x2y2_txs AS (
+    SELECT * FROM aggregator_routed_x2y2_txs_formatted
+    UNION
+    SELECT * FROM direct_x2y2_txs_formated
+    )
+
+SELECT 'ethereum' AS blockchain
+, 'X2Y2' AS project
+, 'v1' AS version
+, txs.block_time
+, txs.token_id
+, txs.collection
+, CASE WHEN currency_contract='0x0000000000000000000000000000000000000000' THEN pu.price*txs.amount_raw/POWER(10, 18)
+    ELSE pu.price*txs.amount_raw/POWER(10, pu.decimals)
+    END AS amount_usd
+, CASE WHEN erct.evt_block_time IS NOT NULL THEN 'erc721'
+    ELSE 'erc1155'
+    END AS token_standard
+, trade_type
+, 1 AS number_of_items
+, CASE WHEN et.`from`=seller THEN 'Offer Accepted'
+    ELSE 'Buy'
+    END AS trade_category
+, 'Trade' AS evt_type
+, txs.seller
+, txs.buyer
+, CASE WHEN currency_contract='0x0000000000000000000000000000000000000000' THEN txs.amount_raw/POWER(10, 18)
+    ELSE txs.amount_raw/POWER(10, pu.decimals)
+    END AS amount_original
+, txs.amount_raw
+, CASE WHEN currency_contract='0x0000000000000000000000000000000000000000' THEN 'ETH'
+    ELSE pu.symbol
+    END AS currency_symbol
+, txs.currency_contract
+, txs.project_contract_address
+, aggregator_name
+, aggregator_address
+, txs.tx_hash
+, et.`from` AS tx_from
+, et.`to` AS tx_to
+, platform_fee_amount_raw
+, CASE WHEN currency_contract='0x0000000000000000000000000000000000000000' THEN platform_fee_amount_raw/POWER(10, 18)
+    ELSE platform_fee_amount_raw/POWER(10, pu.decimals)
+    END AS platform_fee_amount
+, CASE WHEN currency_contract='0x0000000000000000000000000000000000000000' THEN pu.price*platform_fee_amount_raw/POWER(10, 18)
+    ELSE pu.price*platform_fee_amount_raw/POWER(10, pu.decimals)
+    END AS platform_fee_amount_usd
+, platform_fee_percentage
+, royalty_fee_amount_raw
+, CASE WHEN currency_contract='0x0000000000000000000000000000000000000000' THEN royalty_fee_amount_raw/POWER(10, 18)
+    ELSE royalty_fee_amount_raw/POWER(10, pu.decimals)
+    END AS royalty_fee_amount
+, CASE WHEN currency_contract='0x0000000000000000000000000000000000000000' THEN pu.price*royalty_fee_amount_raw/POWER(10, 18)
+    ELSE pu.price*royalty_fee_amount_raw/POWER(10, pu.decimals)
+    END AS royalty_fee_amount_usd
+, royalty_fee_percentage
+, royalty_fee_receive_address
+, CASE WHEN currency_contract='0x0000000000000000000000000000000000000000' THEN 'ETH'
+    ELSE pu.symbol
+    END AS royalty_fee_currency_symbol
+, 'x2y2-' || txs.tx_hash || '-' || txs.token_id || '-' || txs.seller || '-' || txs.evt_index || 'Trade' AS unique_trade_id
+FROM all_x2y2_txs txs
+LEFT JOIN prices.usd pu ON pu.blockchain='ethereum'
+    AND date_trunc('minute', pu.minute)=date_trunc('minute', txs.block_time)
+    AND (pu.contract_address=txs.currency_contract
+        OR (pu.contract_address='0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' AND txs.currency_contract='0x0000000000000000000000000000000000000000'))
+LEFT JOIN ethereum.transactions et ON et.hash=txs.tx_hash
```

Comment:
here, you can add an incremental filter on transactions after line 172:
{% if is_incremental() %}
-- this filter will only be applied on an incremental run
AND tx.block_time >= (select max(block_time) from {{ this }})
{% endif %}